src/schema/script.rs

```diff
@@ -11,234 +11,392 @@
+    #[display("fungible-no-inflation")]
+    FungibleNoInflation = 0x01,
+
+    /// Fungible asset inflation/issue control
+    ///
+    /// Checks that inflation of a fungible asset produces no more than was
+    /// allowed by [`crate::schema::constants::STATE_TYPE_INFLATION_RIGHT`],
+    /// i.e. that the sum of all outputs with
+    /// [`crate::schema::constants::STATE_TYPE_OWNED_AMOUNT`] type is no more
+    /// than that value - plus validates bulletproof data.
+    ///
+    /// Also validates that the sum of the issued asset is equal to the amount
+    /// specified in the [`crate::schema::constants::FIELD_TYPE_ISSUED_SUPPLY`]
+    /// metadata field
+    #[display("fungible-issue")]
+    FungibleIssue = 0x02,
+
+    /// NFT/identity transfer control
+    ///
+    /// Checks that all identities are transferred once and only once, i.e.
+    /// that the _number_ of
+    /// [`crate::schema::constants::STATE_TYPE_OWNED_DATA`] inputs is equal
+    /// to the _number_ of outputs of this type.
+    #[display("nft-transfer")]
+    IdentityTransfer = 0x11,
+
+    /// NFT asset secondary issue control
+    ///
+    /// Checks that inflation of a fungible asset produces no more than was
+    /// allowed by [`crate::schema::constants::STATE_TYPE_INFLATION_RIGHT`],
+    /// i.e. that the sum of all outputs with
+    /// [`crate::schema::constants::STATE_TYPE_OWNED_AMOUNT`] type is no more
+    /// than that value
+    #[display("nft-issue")]
+    NftIssue = 0x12,
+
+    /// Proof-of-burn verification
+    ///
+    /// Currently not implemented in RGBv0 and always validates to TRUE
+    #[display("proof-of-burn")]
+    ProofOfBurn = 0x20,
+
+    /// Proof-of-reserve verification
+    ///
+    /// Currently not implemented in RGBv0 and always validates to TRUE
+    #[display("proof-of-reserve")]
+    ProofOfReserve = 0x21,
+
+    /// Verification of rights splits procedure
+    ///
+    /// Checks that each of the owned rights types were assigned one-to-one
+    /// between inputs and outputs
+    #[display("rights-split")]
+    RightsSplit = 0x30,
 }
 
-#[non_exhaustive]
-#[derive(
-    Clone,
-    Copy,
-    PartialEq,
-    Eq,
-    PartialOrd,
-    Ord,
-    Debug,
-    Display,
-    ToPrimitive,
-    FromPrimitive,
-)]
-#[cfg_attr(
-    feature = "serde",
-    derive(Serialize, Deserialize),
-    serde(crate = "serde_crate", rename_all = "snake_case")
-)]
-#[display(Debug)]
-#[repr(u8)]
-pub enum StandardProcedure {
-    NoInflationBySum = 0x01,
-    FungibleInflation = 0x02,
-    NonfungibleInflation = 0x03,
-    IdentityTransfer = 0x04,
-    ProofOfBurn = 0x10,
-    ProofOfReserve = 0x11,
-    RightsSplit = 0x20,
+impl EmbeddedProcedure {
+    /// Constructs [`EmbeddedProcedure`] from [`EntryPoint`], or returns
+    /// `None` if the provided entry point value does not correspond to any of
+    /// the embedded procedures
+    pub fn from_entry_point(entry_point: EntryPoint) -> Option<Self> {
```

Comment:
It seems `EntryPoint` and `EmbeddedProcedure` are just 1-1 mapping of each other. Is there any specific reason to have a separate `EntryPoint` representation of `EmbededProcedure`? 

I would guess we can use `u32` representation of `EmbeddedProcedure` everywhere we are using `EntryPoint`.

Answering the question: We might need a different `EntryPoint` for non embedded procedures later. Is that the reason?

Reply:
Yes, `EntryPoint` is an offset within the script for starting the VM. Embedded procedures are just use it in a different "hacky" way storing an ID of the embedded procedure in it, but semantically it is different types

Reply:
Understood.